lib/galaxy/tools/filter_failed_collection.xml

```diff
@@ -0,0 +1,16 @@
+<tool id="__FILTER_FAILED_DATASETS__"
+      name="Filter failed"
+      version="1.0.0"
+      tool_type="filter_failed_datasets_collection">
+    <description>from a dataset collection</description>
+    <type class="FilterFailedDatasetsTool" module="galaxy.tools" />
+    <action module="galaxy.tools.actions.model_operations"
+            class="ModelOperationToolAction"/>
+    <inputs>
+        <param type="data_collection" collection_type="list" name="input" label="Input Collection" />
+    </inputs>
+    <outputs>
+        <collection name="output" format_source="input" type_source="input" label="${on_string} (filtered failed datasets)" >
+        </collection>
+    </outputs>
```

Comment:
I can't write useful tests for this tool, it requires datasets in a failed state. I'll add a test to ensure this is identity on okay dataset collections and a note about why it doesn't have more interesting tests as part of the tool XML and a link to the location in ``test_tools.py`` where the more interesting test case is.